django/db/models/lookups.py

```diff
@@ -217,8 +217,6 @@ class BuiltinLookup(Lookup):
     def process_lhs(self, compiler, connection, lhs=None):
         lhs_sql, params = super().process_lhs(compiler, connection, lhs)
         field_internal_type = self.lhs.output_field.get_internal_type()
-        db_type = self.lhs.output_field.db_type(connection=connection)
-        lhs_sql = connection.ops.field_cast_sql(db_type, field_internal_type) % lhs_sql
```

Comment:
We should still call `field_cast_sql()` when it's implemented in the 3rd-party database backends and add a deprecation warning here, e.g.
```python
# RemovedInDjango60Warning.
if connection.ops.field_cast_sql is not BaseDatabaseOperations.field_cast_sql:
    warnings.warn(...)
    db_type = self.lhs.output_field.db_type(connection=connection)
    lhs_sql = connection.ops.field_cast_sql(db_type, field_internal_type) % lhs_sql
```

Reply:
@smithdc1 I can add an extra warning if you're busy :point_up:

Reply:
Yes please. 🎁